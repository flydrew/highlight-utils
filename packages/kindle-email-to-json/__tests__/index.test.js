const toJSON = require("../");
const fs = require("mz/fs");
const path = require("path");

describe("kindle-email-to-json", () => {
  let mail;

  beforeAll(() =>
    fs
      .readFile(path.resolve(__dirname, "../__mocks__/email.txt"), {
        encoding: "utf8"
      })
      .then(contents => {
        mail = contents;
      })
  );

  it("parses book title and authors", () => {
    return toJSON(mail).then(data => {
      expect(data).toMatchSnapshot();
    });
  });
});
