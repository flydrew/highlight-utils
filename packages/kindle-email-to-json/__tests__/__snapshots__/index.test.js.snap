// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kindle-email-to-json parses book title and authors 1`] = `
Object {
  "highlights": Array [
    Object {
      "content": "Information scientists work every day on the design, delegation, and choice of classification systems and standards, yet few see them as artifacts embodying moral and aesthetic choices that in turn craft people’s identities, aspirations, and dignity.",
      "location": "194",
    },
    Object {
      "content": "1. A “standard” is any set of agreed-upon rules for the production of (textual or material) objects. 2. A standard spans more than one community of practice (or site of activity). It has temporal reach as well in that it persists over time. 3. Standards are deployed in making things work together over distance and heterogeneous metrics.",
      "location": "383",
    },
    Object {
      "content": "5. There is no natural law that the best standard shall win—QWERTY, Lotus 123, DOS, and VHS are often cited as examples in this context. The standards that do win may do so for a variety of other reasons: they build on an installed base, they had better marketing at the outset, or they were used by a community of gatekeepers who favored their use. Sometimes standards win due to an outright conspiracy, as in the case of the gas refrigerator documented by Cowan (1985). 6. Standards have significant inertia and can be very difficult and expensive to change.",
      "location": "393",
    },
    Object {
      "content": "these dimensions of standards are in some sense idealized. They embody goals of practice and production that are never perfectly realized, like Plato’s triangles. The process of building to a standardized code, for example, usually includes a face-to-face negotiation between builder(s) and inspector(s), which itself includes a history of relations between those people. Small deviations are routinely overlooked, unless the inspector is making a political point. The idiom “good enough for government use” embodies the common-sense accommodations of the slip between the ideal standard and the contingencies of practice.",
      "location": "408",
    },
    Object {
      "content": "A definition of infrastructure • Embeddedness . Infrastructure is sunk into, inside of, other structures, social arrangements, and technologies, • Transparency . Infrastructure is transparent to use in the sense that it does not have to be reinvented each time or assembled for each task, but invisibly supports those tasks. • Reach or scope . This may be either spatial or temporal—infrastructure has reach beyond a single event or one-site practice; • Learned as part of membership . The taken-for-grantedness of artifacts and organizational arrangements is a sine qua non of membership in a community of practice (Lave and Wenger 1991, Star 1996). Strangers and outsiders encounter infrastructure as a target object to be learned about. New participants acquire a naturalized familiarity with its objects as they become members. • Links with conventions of practice . Infrastructure both shapes and is shaped by the conventions of a community of practice; for example, the ways that cycles of day-night work are affected by and affect electrical power rates and needs. Generations of typists have learned the QWERTY keyboard; its limitations are inherited by the computer keyboard and thence by the design of today’s computer furniture (Becker 1982). • Embodiment of standards . Modified by scope and often by conflicting conventions, infrastructure takes on transparency by plugging into other infrastructures and tools in a standardized fashion. • Built on an installed base . Infrastructure does not grow de novo; it wrestles with the inertia of the installed base and inherits strengths and limitations from that base. Optical fibers run along old railroad lines, new systems are designed for backward compatibility; and failing to account for these constraints may be fatal or distorting to new development processes (Monteiro and Hanseth 1996). • Becomes visible upon breakdown . The normally invisible quality of working infrastructure becomes visible when it breaks: the server is down, the bridge washes out, there is a power blackout. Even when there are backup mechanisms or procedures, their existence further highlights the now visible infrastructure. • Is fixed in modular increments, not all at once or globally . Because infrastructure is big, layered, and complex, and because it means different things locally, it is never changed from above. Changes take time and negotiation, and adjustment with other aspects of the systems involved. 7 Source: Star and Rohleder 1996.",
      "location": "701",
    },
  ],
  "volume": Object {
    "authors": "Bowker, Geoffrey C.; Star, Susan Leigh",
    "title": "Sorting Things Out: Classification and Its Consequences (Inside Technology)",
  },
}
`;
